litestar/config/app.py

```diff
@@ -180,6 +180,11 @@ class AppConfig:
     """
     signature_namespace: dict[str, Any] = field(default_factory=dict)
     """A mapping of names to types for use in forward reference resolution during signature modelling."""
+    signature_types: list[Any] = field(default_factory=list)
```

Comment:
```suggestion
    signature_types: Sequence[Any] = field(default_factory=list)
```

Reply:
As we control the instantiation of the config object:

https://github.com/litestar-org/litestar/blob/b6fc70cf2cda7c42a53bc224622d45d1fd4a304c/litestar/app.py#L319-L368

...then, we know these are concrete collection types, not abstract types.

The advantage of this is for plugins. If this is typed `list[Any]`, a plugin can readily append to it:

```py
config.signature_types.append(PluginManagedType)
```

If typed as `Sequence`:

```py
if not isinstance(config.signature_types, list):
    config.signature_types = list(config.signature_types)
config.signature_types.append(PluginManagedType)
```

Reply:
Aah cool that makes sense.